### DISCUSSION TITLE
About 3-j symbols

### DISCUSSION THREAD

**Discussion by wzm2256** [e3nn/e3nn#405] [Q&A] [answered]
What is the order of the output of the 3j symbols? is it correct?

I test the simplest case: j1=j2=1, J=0

```
o3.wigner_3j(1, 1, 0, dtype=torch.float64)
#tensor([[[0.5774],
#         [0.0000],
#         [0.0000]],

#        [[0.0000],
#        [0.5774],
#         [0.0000]],

#        [[0.0000],
#         [0.0000],
#         [0.5774]]], dtype=torch.float64)
```
On the other hand, I compute all values using the woframe calculator 
https://www.wolframalpha.com/input?i=wigner+3j&assumption=%7B%22F%22%2C+%22Wigner3JCalculator%22%2C+%22m3%22%7D+-%3E%220%22&assumption=%7B%22F%22%2C+%22Wigner3JCalculator%22%2C+%22m2%22%7D+-%3E%221%22&assumption=%7B%22F%22%2C+%22Wigner3JCalculator%22%2C+%22m1%22%7D+-%3E%221%22&assumption=%7B%22C%22%2C+%22wigner+3j%22%7D+-%3E+%7B%22Calculator%22%7D&assumption=%7B%22F%22%2C+%22Wigner3JCalculator%22%2C+%22j1%22%7D+-%3E%221%22&assumption=%7B%22F%22%2C+%22Wigner3JCalculator%22%2C+%22j2%22%7D+-%3E%221%22&assumption=%7B%22F%22%2C+%22Wigner3JCalculator%22%2C+%22j3%22%7D+-%3E%220%22

and here is what I get
|   |   |   |   |
|---|---|---|---|
|       | -1 | 0  | 1  | 
|  -1  |   0  | 0  | 1/sqrt(3)  |  
|   0  |   0   |  -1/sqrt(3)  | 0  |   |
|  1   |     1/sqrt(3) |  0    | 0|

where the rows are m1 and columns are m2. (M=0 of course).

0.5774 is close to 1/sqrt(3), but there are some differences:
1. Is there any change of basis? it seems that there the 1st and the 3rd row/column are swaped. What is the general rule of change of basis?
2. Why the sign is different in the 2nd row?

**Comment by mariogeiger** [comment, marked as answer]
We use the real basis. 

It solves the equation shown in the doc: https://docs.e3nn.org/en/latest/api/o3/o3_wigner.html#e3nn.o3.wigner_3j

Since it's a linear equation, the global scaling is not important.

**Reply by wzm2256** [reply]
Thanks for your reply!
